packages/components/checkbox/src/checkbox.tsx

```diff
@@ -134,16 +134,26 @@ export const Checkbox = forwardRef<CheckboxProps, "input">(function Checkbox(
     onChange,
   })
 
+  const [previousIsChecked, setPreviousIsChecked] = useState(state.isChecked)
+  const [animateIcon, setAnimateIcon] = useState(false)
+
+  if (state.isChecked !== previousIsChecked) {
+    setAnimateIcon(true)
+    setPreviousIsChecked(state.isChecked)
+  }
```

Comment:
Setting state during render is a bit unusual, but apparently it's the recommended way now → https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes.